src/test/test-lwm2m.c

```diff
@@ -562,94 +1322,522 @@ delete_cb(void *data,
+    SOL_LWM2M_RESOURCE_SINGLE_INIT(r, &sec_server[1], SECURITY_OBJECT_IS_BOOTSTRAP,
+        SOL_LWM2M_RESOURCE_DATA_TYPE_BOOL, false);
+    ASSERT(r == 0);
+    SOL_LWM2M_RESOURCE_SINGLE_INT_INIT(r, &sec_server[2], SECURITY_OBJECT_SECURITY_MODE,
+        SOL_LWM2M_SECURITY_MODE_PRE_SHARED_KEY);
+    ASSERT(r == 0);
+    SOL_LWM2M_RESOURCE_SINGLE_INIT(r, &sec_server[3], SECURITY_OBJECT_PUBLIC_KEY_OR_IDENTITY,
+        SOL_LWM2M_RESOURCE_DATA_TYPE_STRING, &sec_server_psk_id);
+    ASSERT(r == 0);
+    SOL_LWM2M_RESOURCE_SINGLE_INIT(r, &sec_server[4], SECURITY_OBJECT_SECRET_KEY,
+        SOL_LWM2M_RESOURCE_DATA_TYPE_STRING, &sec_server_psk_key);
+    ASSERT(r == 0);
+    SOL_LWM2M_RESOURCE_SINGLE_INT_INIT(r, &sec_server[5], SECURITY_OBJECT_SERVER_ID, 102);
+    ASSERT(r == 0);
+
+    printf("DBG: [Bootstrap Write /0/0]---> '%s'\n",
+        sol_lwm2m_bootstrap_client_info_get_name(bs_cinfo));
+    r = sol_lwm2m_bootstrap_server_write(server, bs_cinfo, "/0/0",
+        sec_server, sol_util_array_size(sec_server), write_sec_server_cb, NULL);
+    ASSERT(r == 0);
+
+    for (i = 0; i < sol_util_array_size(sec_server); i++)
+        sol_lwm2m_resource_clear(&sec_server[i]);
+}
+
+static void
+bootstrap_request_cb(void *data,
+    struct sol_lwm2m_bootstrap_server *server,
+    struct sol_lwm2m_bootstrap_client_info *bs_cinfo)
+{
+    int r;
+
+    ASSERT(!strcmp("cli1", sol_lwm2m_bootstrap_client_info_get_name(bs_cinfo)));
+
+    printf("DBG: [Bootstrap Delete /]---> '%s'\n",
+        sol_lwm2m_bootstrap_client_info_get_name(bs_cinfo));
+    r = sol_lwm2m_bootstrap_server_delete_object_instance(server, bs_cinfo, "/",
+        delete_all_cb, NULL);
+    ASSERT(r == 0);
+}
+
+static void
+bootstrap_finish_cb(void *data,
+    struct sol_lwm2m_client *client,
+    enum sol_lwm2m_bootstrap_event event)
+{
+    int r;
+
+    ASSERT(event == SOL_LWM2M_BOOTSTRAP_EVENT_FINISHED);
+
+    r = sol_lwm2m_client_start(client);
+    ASSERT(r == 0);
+}
+#endif
+
 int
 main(int argc, char *argv[])
 {
-    struct sol_lwm2m_server *server;
-    struct sol_lwm2m_client *client;
-    static const struct sol_lwm2m_object *objects[] =
+    struct sol_lwm2m_server *nosec_server;
+    struct sol_lwm2m_client *nosec_client;
+    static const struct sol_lwm2m_object *nosec_objects[] =
     { &security_object, &server_object, &dummy_object, NULL };
+    struct security_obj_instance_ctx *nosec_security_data;
+    struct server_obj_instance_ctx *nosec_server_data;
+    bool nosec_first = true;
     int r;
 
+#ifdef DTLS
+    struct sol_lwm2m_bootstrap_server *bs_server;
+    struct sol_lwm2m_server *sec_server;
+    struct sol_lwm2m_client *sec_client;
+    static const struct sol_lwm2m_object *sec_objects[] =
+    { &security_object, &server_object,
+      &access_control_object, &dummy_object, NULL };
+    struct security_obj_instance_ctx *sec_security_data;
+    struct sol_lwm2m_security_psk *bs_psk, *sec_psk;
+    struct sol_vector bs_known_keys = { }, sec_known_keys = { };
```

Comment:
you can use
`struct sol_vector bs_known_keys = SOL_VECTOR_INIT(struct sol_lwm2m_security_psk);`